addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js

```diff
@@ -3166,10 +3166,22 @@ export class OdooEditor extends EventTarget {
         const files = getImageFiles(ev.clipboardData);
         const targetSupportsHtmlContent = isHtmlContentSupported(sel.anchorNode);
         const clipboardHtml = ev.clipboardData.getData('text/html');
+        // Replace entire link if its label is fully selected.
+        const link = closestElement(sel.anchorNode, 'a', true);
+        if (link && sel.toString() === link.innerHTML.replace(/\u200B/g, '')) {
```

Comment:
I think you're talking about 
```
export function cleanZWS(node) {
    [node, ...descendants(node)]
        .filter(node => node.nodeType === Node.TEXT_NODE)
        .forEach(node => node.nodeValue = node.nodeValue.replace(/\u200B/g, ''));
}
```
Not sure it would be interesting to use it here, as we would have to call it (and change the link's content) before the `if` condition (without the need for `replace` this time).

On the other hand, great think that you made me realize that I meant to call `innerText` here, and not `innerHtml`! Fixing it for the re-push...

Reply:
No I meant `isZWS` but I misread the condition I thought you were checking if the link was entirely made of ZWS. I think it's confusing to have the zws here as it's not really relevant. It just happens that it can get in the way of the checking (and even then, I don't really know why because I guess it should be in `sel.toString()` as well). Wouldn't it work to check if `startOffset === 0` and `endOffset === nodeLength(link)` ?

Reply:
> (and even then, I don't really know why because I guess it should be in sel.toString() as well)

You're right. I have come to a wrong conclusion in my tests because the ZWS was not in the selection - but it could be! The most robust way would be to disregard eventual ZWS on either side:
`sel.toString().replace(/\u200B/g, '') === link.innerHTML.replace(/\u200B/g, '')`
And now it's even worse than before! :sweat_smile: 
I'm pushing with this change, but tell me if you'd rather have me rewrite this some other way...